config/configtls/configtls.go

```diff
@@ -176,6 +176,14 @@ func (c TLSSetting) loadTLSConfig() (*tls.Config, error) {
 		getClientCertificate = func(cri *tls.CertificateRequestInfo) (*tls.Certificate, error) { return certReloader.GetCertificate() }
 	}
 
+	// Setting default TLS minVersion
+	if c.MinVersion == "" {
+		c.MinVersion = "1.0"
+	}
+	// Setting default TLS maxVersion
+	if c.MaxVersion == "" {
+		c.MaxVersion = "1.3"
+	}
```

Comment:
@bogdandrutu your analysis look correct to me. So, essentially you are saying that we want:
- On clients if MinVersion is missing then set it to 1.2.
- On servers if MinVersion is missing then set it to 1.0.
- Both on clients and servers if MaxVersion is missing then set it to 1.3.

Reply:
> Not setting a default value at all crypto/tls will handle the TLS minVersion and maxVersions for us. Should I proceed with updating the PR?

That's a good point. Just pass the zero values and let crypto/tl handle the defaults.

Reply:
I've updated the PR so that we do not add a default via opentelemetry, crypto/tls will handle the default TLS versions.

Reply:
Have we looked into this https://github.com/open-telemetry/opentelemetry-collector/pull/4503 and the CWE-327? Do we want to enforce on the server side "1.2" as well?

Reply:
Was thinking about this problem and I agree that, TLS 1.0 and 1.1 has been deprecated.
https://datatracker.ietf.org/doc/html/rfc8996

However, golang tls package picks the most secure cipher suite for client negotiation; unless the server has **explicitly** set the `MaxVersion` < "1.2" a TLS cipher of TLS1.2 and above will be used for negotiation. 
https://github.com/golang/go/blob/55590f3a2b89f001bcadf0df6eb2dde62618302b/src/crypto/tls/handshake_client.go#L39

For server side, the order of cipher is decided by the client via their preference order. And in this regard, there is a chance that the client has specified a insecure TLS version to be used. https://github.com/golang/go/blob/55590f3a2b89f001bcadf0df6eb2dde62618302b/src/crypto/tls/handshake_server.go#L159.
My opinion is that clients are incharge of the level of secrecy of the data that is being sent. If they are using a client that does not support higher TLS version it is up to them to upgrade it. Not the server's duty to force secrecy but rather be as compatible as possible.

That being said, the above is all my opinion and I'm open in changing it.

Reply:
@tigrannajaryan @codeboten please tell me if it is fine to downgrade the server min version.

Reply:
My understand is that this PR is about fixing the defaults to match the default of what the Go's standard library defines, so let's keep it that way. The code in this PR looks correct to me as it is.

If we want to require higher version number than Go's standard library then it is best to discuss it in in a separate issue/PR.

Reply:
This change just delegates to the golang compiler/vm to pick the min/max TLS. Am I wrong? We are already building the binaries with golang 1.8 (see https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/.github/workflows/ci-goreleaser.yaml#L35), but we have "go 1.17" in the go.mod. Does that mean that the min version is 1.0?

Reply:
Oh, I didn't realize we were already using Go 1.18 on the official binaries. Then I believe regardless of what `go.mod` says, this will not mean a change on the official binaries, so I feel like doing this is okay

Reply:
Just rebased the PR to `main`, is there anything blocking this PR now? I'll be happy to follow through with the changes.